templates/expo-template-bare-minimum/android/build.gradle

```diff
@@ -4,10 +4,13 @@ import org.apache.tools.ant.taskdefs.condition.Os
 
 buildscript {
     ext {
-        buildToolsVersion = "31.0.0"
+        buildToolsVersion = findProperty('android.buildToolsVersion') ?: "31.0.0"
         minSdkVersion = 21
-        compileSdkVersion = 31
-        targetSdkVersion = 31
+        compileSdkVersion = Integer.parseInt(findProperty('android.compileSdkVersion') ?: 31)
+        targetSdkVersion = Integer.parseInt(findProperty('android.targetSdkVersion') ?: 31)
+        if (hasProperty('android.kotlinVersion')) {
+            kotlinVersion = findProperty('android.kotlinVersion')
```

Comment:
cool. Makes me think we should document how other plugins should interact with this. In the past, other plugins would just use the values in the app.json schema. Perhaps it would be by importing `expo-build-properties`?

Another thing worth noting is that plugins often set the 'minimum required version' so they don't accidentally set a lower version than what the user has set. [reference](https://github.com/expo/config-plugins/blob/31aee40ee3cb225990ae322e3d4c6043687ed5cd/packages/react-native-webrtc/src/withWebRTC.ts#L35-L36)

Reply:
> cool. Makes me think we should document how other plugins should interact with this. In the past, other plugins would just use the values in the app.json schema. Perhaps it would be by importing `expo-build-properties`?

to be honest, i don't have a good solution in mind. all the properties are not officially supported in app.json schema. so i think there's no way to read the properties even in the past. importing `expo-build-properties` to read the properties seems not feasible because we removed the `createRunOncePlugin`, other config plugins maybe reuse this plugin and pass different config by their own.

> Another thing worth noting is that plugins often set the 'minimum required version' so they don't accidentally set a lower version than what the user has set. [reference](https://github.com/expo/config-plugins/blob/31aee40ee3cb225990ae322e3d4c6043687ed5cd/packages/react-native-webrtc/src/withWebRTC.ts#L35-L36)

minimal version check added in https://github.com/expo/expo/pull/17106/commits/56054980afca4f4767fac0b5dc66372dbadad823

> I worry packages will no longer be able to automatically set the version if they can't read it. So we may also just need to document that config plugins should no longer set these versions and instead users should update versions locally themselves. This seems a bit more intentional and correct, but it could get messy if the user has to juggle a bunch of versions manually.

my hypothesis is that existing config plugins should use dangerous mods to change the values in templates. maybe the changes here could be claimed a breaking change from sdk 45?